StoreKitTests/StoreKitTests.swift

```diff
@@ -46,9 +55,221 @@ class StoreKitTests: XCTestCase {
+            completionCalled = true
+            expect(purchaserInfo?.entitlements.all.count) == 0
+            expect(created).to(beFalse())
+            expect(logInError).to(beNil())
+        }
+
+        expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
+
+        Purchases.shared.restoreTransactions()
+
+        waitUntilPurchaseGoesThrough()
+    }
+
+    func testPurchaseAsIdentifiedThenLogOutThenRestoreGrantsEntitlements() {
+        configurePurchases()
+        var completionCalled = false
+        let existingUserID = UUID().uuidString
+        expect(self.purchasesDelegate.purchaserInfoUpdateCount).toEventually(equal(1), timeout: .seconds(10))
+
+        Purchases.shared.logIn(existingUserID) { logInPurchaserInfo, created, logInError in
+            self.purchaseMonthlyOffering()
+            completionCalled = true
+        }
+
+        expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
+
+        waitUntilPurchaseGoesThrough()
+
+        completionCalled = false
+
+        Purchases.shared.logOut { purchaserInfo, error in
+            expect(purchaserInfo?.entitlements.all.count) == 0
+            expect(error).to(beNil())
+            completionCalled = true
+        }
+
+        expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
+
+        Purchases.shared.restoreTransactions()
+
+        waitUntilPurchaseGoesThrough()
+    }
+
+    func testLogInReturnsCreatedTrueWhenNewAndFalseWhenExisting() {
+        configurePurchases()
+
+        let anonUserID = Purchases.shared.appUserID
+        let identifiedUserID = "\(#function)_\(anonUserID)".replacingOccurrences(of: "RCAnonymous", with: "")
+
+        var completionCalled = false
+        Purchases.shared.logIn(identifiedUserID) { identifiedPurchaserInfo, created, error in
+            expect(error).to(beNil())
+            expect(created).to(beTrue())
+            Purchases.shared.logOut { loggedOutPurchaserInfo, logOutError in
+                Purchases.shared.logIn(identifiedUserID) { identifiedPurchaserInfo, created, error in
+                    expect(error).to(beNil())
+                    expect(created).to(beFalse())
+                    completionCalled = true
+                }
+            }
+        }
+
+        expect(completionCalled).toEventually(beTrue(), timeout: .seconds(10))
+    }
+
+    func testLogInThenLogInAsAnotherUserWontTransferPurchases() {
+        configurePurchases()
+
+        let userID1 = UUID().uuidString
+        let userID2 = UUID().uuidString
+
+        Purchases.shared.logIn(userID1) { identifiedPurchaserInfo, created, error in
+            self.purchaseMonthlyOffering()
+        }
+
+        waitUntilPurchaseGoesThrough()
+
+        testSession.clearTransactions()
+
+        Purchases.shared.logIn(userID2) { identifiedPurchaserInfo, created, error in
+            expect(identifiedPurchaserInfo?.entitlements.all.count) == 0
```

Comment:
I think this could be extracted to a method since it's been done in a bunch of places :)

Reply:
done!